### ISSUE TITLE
EditButton & ShowButton not working on first run

### ISSUE THREAD

**Issue by wesley6j** [marmelab/admin-on-rest#665]
The condition for shouldUpdate does not work the first time after a refresh:
```
shouldUpdate((props, nextProps) =>
    props.record
    && props.record.id !== nextProps.record.id
    || props.basePath !== nextProps.basePath
),
```
If a resource has both show and edit page and I open the show page after a refresh. `props.record` was undefined before the data load, and  equal to `nextProps.record` afterwards. Therefore, the link will be something like `posts/undefined`.

Note that this bug only occurs after a refresh.

**Comment by wesley6j** [comment]
Condition should be like this:
```
shouldUpdate((props, nextProps) =>
    !props.record || (props.record
    && props.record.id !== nextProps.record.id
    || props.basePath !== nextProps.basePath)
),
```

**Comment by fzaninotto** [comment]
For the record, why did you close your own issue?

**Comment by wesley6j** [comment]
Hi, @fzaninotto . I made a [PR](https://github.com/marmelab/admin-on-rest/pull/671) for this issue. 

I have another question related to this issue: 

Should `shouldUpdate` be applied to `DeleteButton` as well?

**Comment by fzaninotto** [comment]
Fixed by #671
